admin/spam/views.py

```diff
@@ -174,6 +174,8 @@ def form_valid(self, form):
         try:
             if int(form.cleaned_data.get('confirm')) == Comment.SPAM:
                 item.confirm_spam(save=True)
+                item.is_deleted = True
```

Comment:
My vote would be moving towards a generalized way of indicating deletion, but this is fine for now.